deliver/lib/deliver/upload_screenshots.rb

```diff
@@ -10,20 +10,21 @@ def upload(options, screenshots)
       UI.user_error!("Could not find a version to edit for app '#{app.name}'") unless v
 
       UI.message("Starting with the upload of screenshots...")
+      screenshots_per_language = screenshots.group_by(&:language)
 
-      # First, clear all previously uploaded screenshots, but only where we have new ones
-      # screenshots.each do |screenshot|
-      #   to_remove = v.screenshots[screenshot.language].find_all do |current|
-      #     current.device_type == screenshot.device_type
-      #   end
-      #   to_remove.each { |t| t.reset! }
-      # end
-      # This part is not working yet...
+      if options[:overwrite_screenshots]
+        UI.message("Removing all previously uploaded screenshots...")
+        # First, clear all previously uploaded screenshots
+        screenshots_per_language.keys.each do |language|
+          v.screenshots[language].each_with_index do |t, index|
+            v.upload_screenshot!(nil, index, t.language, t.device_type)
```

Comment:
Yes. I tried this method following ways.

1. Add 4.7 and 5.5 inch screenshots on Finder
2. Run deliver
3. Remove the 4.7 inch screenshot
4. Run deliver with `overwrite_screenshots` option
5. the 4.7 inch screenshot was removed on iTunes Connect

Then it seems work well.